polygeist/include/mlir/Dialect/Polygeist/Utils/TransformUtils.h

```diff
@@ -230,6 +237,66 @@ class LoopTools {
                    const LoopVersionCondition &versionCond) const;
 };
 
+//===----------------------------------------------------------------------===//
+// VersionConditionBuilder
+//===----------------------------------------------------------------------===//
+
+using AccessorType = TypedValue<MemRefType>;
+using AccessorPairType = std::pair<AccessorType, AccessorType>;
+class VersionConditionBuilder {
```

Comment:
Add comment to this class to explain its purpose please.

Reply:
done.